extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/impl/ElasticProcessor.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic.impl;
+
+import com.hazelcast.function.ConsumerEx;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.Traversers;
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.elastic.ElasticSourceBuilder;
+import com.hazelcast.logging.ILogger;
+import org.apache.http.HttpHost;
+import org.apache.lucene.search.TotalHits;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.search.ClearScrollRequest;
+import org.elasticsearch.action.search.ClearScrollResponse;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchScrollRequest;
+import org.elasticsearch.client.Node;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.slice.SliceBuilder;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singleton;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+final class ElasticProcessor<T> extends AbstractProcessor {
```

Comment:
what is the reason here we need a custom processor rather than using the source builder? We also use a different naming convention for processors. (look at StreamKafkaP etc)

Reply:
In the `ElasticProcessorMetaSupplier` the list of shards is loaded and assigned to nodes.

The shards can't be loaded when the source is created, the host from which you submit may not have direct network access to elastic (the cluster).

We could load the shards during initialization of the context of each processor in the builder, but there is a small chance that different processors would get different view of the shards (some might start relocating, disconnect etc..).

I also followed the structure of the hadoop connector.